### PR TITLE
Alpine linux compile fix

### PR DESCRIPTION
**Pull request by mrakh**
Some simple changes to fix warnings/errors when compiling on Alpine Linux. To reproduce the issue:

```bash
docker run --rm -it alpine
apk add git g++ make linux-headers
git clone https://github.com/axboe/liburing
cd liburing && make
```

### CODE REVIEW

**Comment by axboe** [comment]
Can you rebase this to get rid of the merge? Seems odd and unnecessary.

**Comment by ammarfaizi2** [comment]
@mrakh also use `Signed-off-tag` in your commit.

File: test/io_uring_register.c [hunk 1]
```diff
@@ -628,7 +628,7 @@ main(int argc, char **argv)
 		return 1;
 	}
 	mlock_limit = rlim.rlim_cur;
-	printf("RELIMIT_MEMLOCK: %lu (%lu)\n", rlim.rlim_cur, rlim.rlim_max);
+	printf("RELIMIT_MEMLOCK: %llu (%llu)\n", rlim.rlim_cur, rlim.rlim_max);
 	devnull = open("/dev/null", O_RDWR);
 	if (devnull < 0) {
```
Comment (ammarfaizi2) [inline test/io_uring_register.c:633]: This is broken, you need a cast for this.

```
io_uring_register.c: In function ‘main’:
io_uring_register.c:631:30: error: format ‘%llu’ expects argument of type ‘long long unsigned int’, but argument 2 has type ‘rlim_t’ {aka ‘long unsigned int’} [-Werror=format=]
  631 |  printf("RELIMIT_MEMLOCK: %llu (%llu)\n", rlim.rlim_cur, rlim.rlim_max);
      |                           ~~~^            ~~~~~~~~~~~~~
      |                              |                |
      |                              |                rlim_t {aka long unsigned int}
      |                              long long unsigned int
      |                           %lu
io_uring_register.c:631:36: error: format ‘%llu’ expects argument of type ‘long long unsigned int’, but argument 3 has type ‘rlim_t’ {aka ‘long unsigned int’} [-Werror=format=]
  631 |  printf("RELIMIT_MEMLOCK: %llu (%llu)\n", rlim.rlim_cur, rlim.rlim_max);
      |                                 ~~~^                     ~~~~~~~~~~~~~
      |                                    |                         |
      |                                    long long unsigned int    rlim_t {aka long unsigned int}
      |                                 %lu
gcc -Werror -D_GNU_SOURCE -D__SANE_USERSPACE_TYPES__ -I../src/include/ -include ../config-host.h -g -O2 -Wall -Wextra -Werror -Wno-unused-parameter -Wno-sign-compare -Wstringop-overflow=0 -Warray-bounds=0 -DLIBURING_BUILD_TEST -o io_uring_setup io_uring_setup.c helpers.o ../src/syscall.o  -L../src/ -luring
cc1: all warnings being treated as errors
make[1]: *** [Makefile:1[99](https://github.com/ammarfaizi2/liburing/runs/5262881055?check_suite_focus=true#step:4:99): io_uring_register] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/home/runner/work/liburing/liburing/test'
make: *** [Makefile:11: all] Error 2
Error: Process completed with exit code 2.
```

**Review by ammarfaizi2 — CHANGES_REQUESTED** [review]

File: test/io_uring_register.c [hunk 2]
```diff
@@ -628,7 +628,7 @@ main(int argc, char **argv)
 		return 1;
 	}
 	mlock_limit = rlim.rlim_cur;
-	printf("RELIMIT_MEMLOCK: %lu (%lu)\n", rlim.rlim_cur, rlim.rlim_max);
+	printf("RELIMIT_MEMLOCK: %llu (%llu)\n", rlim.rlim_cur, rlim.rlim_max);
```
Comment (ammarfaizi2) [inline test/io_uring_register.c:631]:
```suggestion
	printf("RELIMIT_MEMLOCK: %llu (%llu)\n", (unsigned long long) rlim.rlim_cur, 
		(unsigned long long) rlim.rlim_max);
```

**Comment by mrakh** [comment]
The bad merge, unverified commit and missing `Signed-off-by` tag should be fixed now.

File: test/io_uring_register.c (hunk 2 above)
Comment (mrakh) [reply test/io_uring_register.c]: Added.

**Review by ammarfaizi2 — APPROVED** [review]

**Pull request merged**